api-gateway/src/main/java/com/petclinic/bffapigateway/dtos/Auth/TokenResponseDTO.java

```diff
@@ -0,0 +1,21 @@
+package com.petclinic.bffapigateway.dtos.Auth;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder(toBuilder = true)
+public class TokenResponseDTO {
+
+        private String token;
+        private String userId;
+        private String email;
+        private List<String> roles;
+
+}
```

Comment:
Why did you create a token DTO exactly?

Reply:
To reduce the numbers of call since like that we make sure we have everything in one request.